From a60939d35073c382bf0929fff4f053f77e4da864 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@amd.com>
Date: Fri, 7 Oct 2022 14:03:37 -0500
Subject: [PATCH] add an rpath to things not compiled into a library

A harmless error shows up in debian packages at build time:
```
dpkg-shlibdeps: warning: cannot find library libfwupdplugin.so needed by debian/fwupd/usr/lib/x86_64-linux-gnu/fwupd-1.8.6/libfu_plugin_flashrom.so (ELF format: 'elf64-x86-64' abi: '0201003e00000000'; RPATH: '')
```

This doesn't cause a functional problem because libfwupdplugin has already
been loaded by the daemon by the time these libraries are loaded.

In case the `dpkg-shlibdeps` checker becomes more stringent in the future
fix the warning.
---
 plugins/flashrom/meson.build      | 1 +
 plugins/modem-manager/meson.build | 1 +
 src/meson.build                   | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/plugins/flashrom/meson.build b/plugins/flashrom/meson.build
index a859cd818..269fa62fb 100644
--- a/plugins/flashrom/meson.build
+++ b/plugins/flashrom/meson.build
@@ -12,6 +12,7 @@ shared_module('fu_plugin_flashrom',
   ],
   include_directories: plugin_incdirs,
   install: true,
+  install_rpath: libdir_pkg,
   install_dir: libdir_pkg,
   link_with: plugin_libs,
   c_args: [
diff --git a/plugins/modem-manager/meson.build b/plugins/modem-manager/meson.build
index 12396b689..325c55e05 100644
--- a/plugins/modem-manager/meson.build
+++ b/plugins/modem-manager/meson.build
@@ -25,6 +25,7 @@ shared_module('fu_plugin_modem_manager',
   ],
   include_directories: plugin_incdirs,
   install: true,
+  install_rpath: libdir_pkg,
   install_dir: libdir_pkg,
   c_args: cargs,
   link_with: plugin_libs,
diff --git a/src/meson.build b/src/meson.build
index 84c84c986..ebb262224 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -86,6 +86,7 @@ fwupdutil = library(
     systemd_src,
   ],
   install: true,
+  install_rpath: libdir_pkg,
   install_dir: libdir_pkg,
   include_directories: [
     root_incdir,
@@ -197,6 +198,7 @@ fwupdengine = library(
   resources_src,
   sources: fwupd_engine_src,
   install: true,
+  install_rpath: libdir_pkg,
   install_dir: libdir_pkg,
   include_directories: plugin_incdirs,
   dependencies: [
-- 
2.34.1

